foundations/javascript_basics/fundamentals-5.md

```diff
@@ -28,47 +28,116 @@ As an example of what we mean, let's consider a `sumOfTripledEvens` function. It
+These methods are slightly more complicated than you've been used to, so let's take a moment to understand how to use them.
+
+#### The .map() method
+
+`.map(callback)` is one such function. It expects a `callback` as an argument, which is a fancy way to say "I want to pass another function as an argument to my function".
+
+Let's say we had a function `addOne(num)`, which takes in `num` as an argument and outputs that `num` increased by 1. 
+And let's say we had an array of numbers, `const arr = [1, 2, 3, 4, 5]` and we'd like to increase all of these numbers by `+1` using our `addOne(num)` function.
+Instead of making a `for` loop and iterating over the above array, we could use our `.map(callback)` array method instead, which **automatically** iterates over an array for us. 
+We don't need to do any extra work aside from simply passing the function we want to use in:
+```js
+function addOne(num) {
+  return ++num;
 }
+const arr = [1, 2, 3, 4, 5];
+arr.map(addOne); // Outputs [2, 3, 4, 5, 6]
 ```
 
-In the above function we use 3 array methods that accomplish the same task. Let's assume we input the array `[4, 5, 1, 10, 2, 5, 9]`. To break it down:
+This is a much more elegant approach, what do you think? We could even go a step further and define a function right inside of `map()` like so:
 
-- `.filter` will take our `array` and remove all non-even numbers from it, so it the result will be `[4, 10, 2]`.
-- `.map` will take the result of `.filter` and then transform all those numbers into triple numbers, it's result will be `[12, 30, 6]`.
-- `.reduce` will take the result of `.map` and convert that array to a single value. The way it does this is by storing the sum in the `accumulator` variable, which we initialise to `0` as the second argument of the function. Then, it will iterate through the array returned by `.map` and add every element to `accumulator`, finally returning the accumulator.
-- Finally, it returns the output of `.reduce` from the function.
+```js
+const arr = [1, 2, 3, 4, 5];
+arr.map(num => num + 1); // Outputs [2, 3, 4, 5, 6]; 
+```
+
+Can you try think of a way to multiply every number in the array by 3 instead of adding 1 to it?  
+We could do it like this:
+```js
+arr.map(num => num * 3); // Outputs [6, 9, 12, 15, 18];
+```
 
-In the end, we have a function that is more concise, readable and less prone to bugs.
+#### The .filter() method
+
+`.filter(callback)` is very similar to `.map(callback)`, it still iterates through the array and applies the callback function on every item. However, now instead of simply transforming values, it evaluates them. 
+Let's say we had a function, `isOdd` that returns either `true` if a number is odd or `false` if it isn't. 
+
+The `.filter(callback)` expects the `callback` to return either `true` or `false`. If it returns `true`, the value is included in the output. Otherwise, it isn't.
```

Comment:
```suggestion
The `filter` method expects the `callback` to return either `true` or `false`. If it returns `true`, the value is included in the output. Otherwise, it isn't.
```